src/services/storage/drivers/postgres/index.ts

```diff
@@ -66,6 +76,29 @@ export class PostgresStorage implements StorageDriver {
 		])
 		await this.db.schema.dropSchemaIfExists(this.config.appName)
 	}
+
+	public async pushUpdates(): Promise<void> {
+		//run over the cache and dispatch db pushes
+		for (const [id, count] of Object.entries(this.countersCache.infoCount)) {
+			await this.db
+				.table<UrlInformation>('url_information')
+				.where('url_id', '=', id)
+				.increment('info_visit_count', count)
+				.update({ lastUsed: new Date().toISOString() })
+		}
+		for (const [id, count] of Object.entries(this.countersCache.visitCount)) {
+			await this.db
+				.table<UrlInformation>('url_information')
+				.where('url_id', '=', id)
+				.increment('url_visit_count', count)
+		}
```

Comment:
It's better to just dispatch a single request for it. Something like [this](https://stackoverflow.com/a/11564089) should work, but there might be more efficient solutions. On that note, it'll probably be better to change the counters structure to { [id: string]: { visitCount: number, infoCount: number } }, so that you can update the row once (this is probably true for most DBs anyway).

Also, we should check that the previous update is done before running the next.

P.S even if we did want to send a different request for every update, this is most definitely not the best way since you `await` the request before starting the next one, which means you're running it in series instead of parallel.

Regarding the P.S: This is an example of a more efficient way (**not relevant to our case**)
```suggestion
		const infoCountPromises = Object.entries(this.countersCache.infoCount).map(([id, count])=>this.db
				.table<UrlInformation>('url_information')
				.where('url_id', '=', id)
				.increment('info_visit_count', count)
				.update({ lastUsed: new Date().toISOString() }))
		const visitCountPromises = Object.entries(this.countersCache.visitCount).map(([id, count]) => this.db
				.table<UrlInformation>('url_information')
				.where('url_id', '=', id)
				.increment('url_visit_count', count))
				
		await Promise.all([...infoCountPromises, ...visitCountPromises])
```